myriad-scheduler/src/main/java/com/ebay/myriad/api/ClustersResource.java

```diff
@@ -96,6 +101,48 @@ public Response flexUp(FlexUpClusterRequest request) {
 
     @Timed
     @PUT
+    @Path("/flexupservice")
+    @Produces(MediaType.TEXT_PLAIN)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response flexUpservice(FlexUpServiceRequest request) {
+      Preconditions.checkNotNull(request,
+              "request object cannot be null or empty");
+
+      LOGGER.info("Received Flexup a Service Request");
+
+      Integer instances = request.getInstances();
+      String profile = request.getServiceName();
+
+      LOGGER.info("Instances: {}", instances);
+      LOGGER.info("Profile: {}", profile);
+      
+      // Validate profile request
+      Response.ResponseBuilder response = Response.status(Response.Status.ACCEPTED);
+      
+      if (cfg.getServiceConfiguration(profile) == null) {
+        response.status(Response.Status.BAD_REQUEST)
+                .entity("Service does not exist: " + profile);
+        LOGGER.error("Provided service does not exist " + profile);
+        return response.build();
+      }
+      
+      if (!this.isValidInstanceSize(instances)) {
+        response.status(Response.Status.BAD_REQUEST)
+                .entity("Invalid instance size: " + instances);
+        LOGGER.error("Invalid instance size request " + instances);
+        return response.build();
+      }
+
+      try {
+        this.myriadOperations.flexUpAService(instances, profile);
+      } catch (MyriadBadConfigurationException e) {
```

Comment:
The current flexup/down API implementation is first validating the API request before invoking the corresponding method on MyriadOperations. I suggest we stay consistent with that and not introduce a new MyriadBadConfigurationException.